cogs/strike.py

```diff
@@ -698,6 +699,15 @@ async def on_member_update(self, before: discord.Member, after: discord.Member)
         if not after.timed_out or before.timed_out == after.timed_out:
             return
 
+        audit_log_entry: discord.AuditLogEntry
+        async for audit_log_entry in main_guild.audit_logs(limit=5):
+            audit_log_action: discord.AuditLogAction = audit_log_entry.action
+            if "auto_moderation_user_communication_disabled" in str(audit_log_action):
```

Comment:
Unnecessary casting to string & temporary variable. Use enum comparison check. (Use boolean temporary variable to prevent line-length linting issues)

Fix:
```python
async for ...:
    AUDIT_LOG_ENTRY_IS_AUTOMOD_ACTION: bool = audit_log_entry.action == (
        discord.AuditLogAction.auto_moderation_user_communication_disabled
    )
    if AUDIT_LOG_ENTRY_IS_AUTOMOD_ACTION:
        ...
```